pkg/build/builder/common.go

```diff
@@ -82,3 +84,33 @@ func resetHTTPProxy(originalProxies map[string]string) {
 		os.Setenv("https_proxy", proxy)
 	}
 }
+
+func updateBuildRevision(c client.BuildInterface, build *api.Build, sourceInfo *s2iapi.SourceInfo) {
+	if build.Spec.Revision != nil {
+		return
+	}
+	build.Spec.Revision = &api.SourceRevision{
+		Type: api.BuildSourceGit,
+		Git: &api.GitSourceRevision{
+			Commit:  sourceInfo.CommitID,
+			Message: sourceInfo.Message,
+			Author: api.SourceControlUser{
+				Name:  sourceInfo.AuthorName,
+				Email: sourceInfo.AuthorEmail,
+			},
+			Committer: api.SourceControlUser{
+				Name:  sourceInfo.CommitterName,
+				Email: sourceInfo.CommitterEmail,
+			},
+		},
+	}
+
+	// Reset ResourceVersion to avoid a conflict with other updates to the build
+	build.ResourceVersion = ""
+
+	glog.V(4).Infof("Setting build revision to %#v", build.Spec.Revision.Git)
+	_, err := c.UpdateDetails(build)
+	if err != nil {
+		glog.Warningf("An error occurred saving build revision: %v", err)
```

Comment:
Hmm, In some cases yes. (I can think of the new-app code) In this case, however, I didn't see much benefit in having the calling code in both the STI and Docker builder check for the error and log the warning. If you feel strongly about it, I can refactor, but to me this is better. Also, it's not meant for general use, it's mainly there to abstract out common code in both builders.